src/components/PinInput.tsx

```diff
@@ -1,23 +1,37 @@
+import { StackNavigationProp } from '@react-navigation/stack';
 import React, { PureComponent } from 'react';
 import { StyleSheet, View } from 'react-native';
 import { CodeField } from 'react-native-confirmation-code-field';
 
-import { CONST } from 'app/consts';
+import { CONST, MainCardStackNavigatorParams, Route } from 'app/consts';
+import { noop } from 'app/helpers/helpers';
 import { palette } from 'app/styles';
 
+type NavigationType =
+  | StackNavigationProp<MainCardStackNavigatorParams, Route.CreatePin>
+  | StackNavigationProp<MainCardStackNavigatorParams, Route.CurrentPin>;
+
 interface Props {
   value: string;
   onTextChange: (pin: string) => void;
+  navigation: NavigationType;
 }
 
 export class PinInput extends PureComponent<Props> {
   codeFieldRef = React.createRef<any>();
+  onFocus: Function = noop;
 
   componentDidMount() {
     // workaround for autoFocus prop issues
-    setTimeout(() => {
-      this.focus();
-    }, 500);
+    this.onFocus = this.props.navigation.addListener('focus', () => {
```

Comment:
what about removing this listener?

Reply:
> what about removing this listener?

It's removed in `componentWillUnmount` function.
https://reactnavigation.org/docs/navigation-events/

Reply:
ok I see, could you name it unsubscribeOnFocus?

Reply:
> ok I see, could you name it unsubscribeOnFocus?

maybe just `focusListener` to be sure that the variable is a listener?

Reply:
I would like to see that this is a function to unsubscribe, so maybe `unsubscribeFocusListener`?

Reply:
ok, changed